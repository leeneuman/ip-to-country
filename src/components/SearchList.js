import React from 'react';
import IpItem from './ipItem';

function SearchList({ itemsList = [] }) {
	if (itemsList.length === 0) return null;

	return (
		<div data-testid="search-list" className="mt-3 pt-3" style={{ borderTop: '1px solid #dee2e6' }}>
			{itemsList.map((i) => <IpItem key={`ip-item-${i}`} item={i} />)}
		</div>
	);
}

export default SearchList;
