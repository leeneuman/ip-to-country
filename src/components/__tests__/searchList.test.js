import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import SearchList from '../searchList';

afterEach(cleanup);

test('should render serach list component', () => {
	const itemsList = [ 1, 2 ];

	render(<SearchList itemsList={itemsList} />);
	const itemElement = screen.getByTestId('search-list');
	expect(itemElement).toBeInTheDocument();
});

test('matches snapshot', () => {
	const itemsList = [ 1, 2 ];

	const tree = renderer.create(<SearchList itemsList={itemsList} />).toJSON();
	expect(tree).toMatchSnapshot();
});
