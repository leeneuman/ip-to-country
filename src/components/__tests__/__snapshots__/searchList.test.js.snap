// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="mt-3 pt-3"
  data-testid="search-list"
  style={
    Object {
      "borderTop": "1px solid #dee2e6",
    }
  }
>
  <div
    className="d-flex p-2 flex-column"
    data-testid="item-1"
  >
    <div
      className="d-flex align-items-center"
    >
      <div
        className="d-flex align-items-center justify-content-center mr-3"
        style={
          Object {
            "backgroundColor": "#e5e5e5",
            "borderRadius": "50%",
            "flexShrink": "0",
            "height": "30px",
            "width": "30px",
          }
        }
      >
        1
      </div>
      <input
        aria-label="ip address"
        className="form-control mr-3 w-50"
        data-testid="item-input-1"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="d-flex p-2 flex-column"
    data-testid="item-2"
  >
    <div
      className="d-flex align-items-center"
    >
      <div
        className="d-flex align-items-center justify-content-center mr-3"
        style={
          Object {
            "backgroundColor": "#e5e5e5",
            "borderRadius": "50%",
            "flexShrink": "0",
            "height": "30px",
            "width": "30px",
          }
        }
      >
        2
      </div>
      <input
        aria-label="ip address"
        className="form-control mr-3 w-50"
        data-testid="item-input-2"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
